File: sdk/src/main/java/com/cxense/cxensesdk/ConsentOption.java
package com.cxense.cxensesdk;

import androidx.annotation.NonNull;

public enum ConsentOption {
    CONSENT_REQUIRED("y"),
    PV_ALLOWED("pv"),
    RECS_ALLOWED("recs"),
    SEGMENT_ALLOWED("segment"),
    AD_ALLOWED("ad");

    private String value;

    ConsentOption(@NonNull String value) {
        this.value = value;
    }

    @NonNull
    public String getValue() {
        return value;
    }
}

File: sdk/src/main/java/com/cxense/cxensesdk/model/ConsentOption.kt
package com.cxense.cxensesdk.model

/**
 * User consent options
 */
@Suppress("unused", "MemberVisibilityCanBePrivate") // Public API.
enum class ConsentOption(val value: String) {
    /**
     * Explicit consent from user is required before processing data
     */
    CONSENT_REQUIRED("y"),
    /**
     * User allowed Page view tracking, DMP event tracking and browsing habit collection to understand a user’s interests and profile.
     */
    PV_ALLOWED("pv"),
    /**
     * User allowed personalisation of content recommendations and suggested content based on user interests and browsing habits.
     */
    RECS_ALLOWED("recs"),
    /**
     * User allowed audience segmentation - processing of browsing habits and first party data to include users in specific audience segments.
     */
    SEGMENT_ALLOWED("segment"),
    /**
     * User allowed targeting advertising based on browsing habits and audience segmentation.
     */
    AD_ALLOWED("ad")
}
